frontends/election-manager/src/components/smartcard_modal/card_details_view.tsx

```diff
@@ -103,70 +111,72 @@ export function CardDetailsView({
-      {/* An empty div to maintain space between the header and subsequent p. TODO: Consider adding
-        a `maintainSpaceBelowHeaders` prop to `Prose` */}
-      <div />
+    <Prose textCenter theme={fontSizeTheme.medium}>
+      <h1>{userRoleToReadableString(role)} Card</h1>
+      {role !== 'superadmin' && <p>{electionDisplayString}</p>}
+
+      <HorizontalRule />
       {actionStatus && (
-        <StatusMessage
-          actionStatus={actionStatus}
-          programmedUser={programmedUser}
-        />
+        <StatusMessageContainer large={actionStatus.status === 'Success'}>
+          <StatusMessage
+            actionStatus={actionStatus}
+            programmedUser={programmedUser}
+          />
+        </StatusMessageContainer>
       )}
-      <CardDetailsTable>
-        <thead>
-          <tr>
-            <th>Role</th>
-            <th>Election</th>
-          </tr>
-        </thead>
-        <tbody>
-          <tr>
-            <td>{userRoleToReadableString(role)}</td>
-            <td>{electionDisplayString}</td>
-          </tr>
-        </tbody>
-      </CardDetailsTable>
-      <p>Remove card to leave this screen.</p>
-      {'passcode' in programmedUser &&
-        (role === 'superadmin' ||
-          // If the card is from a prior election, no need to display PIN resetting. Unprogramming is
-          // the only meaningful action in this case
-          doesCardElectionHashMatchMachineElectionHash) && (
-          <p>
-            <Button disabled={cardJustProgrammed} onPress={resetCardPin}>
-              Reset Card PIN
-            </Button>
-          </p>
-        )}
-      {/* Don't allow unprogramming super admin cards to ensure election officials don't get
-        accidentally locked out. Likewise prevent unprogramming when there's no election definition
-        on the machine since cards can't be programmed in this state */}
-      {(role === 'admin' || role === 'pollworker') && electionDefinition && (
-        <React.Fragment>
-          <p>
+      {!showingSuccessOrErrorMessage && possibleActions.size > 0 && (
+        <p>
+          {possibleActions.has('PinReset') && (
+            <Button onPress={resetCardPin}>Reset Card PIN</Button>
+          )}
+          {possibleActions.size > 1 && ' '}
+          {possibleActions.has('Unprogram') && (
             <Button
               danger={doesCardElectionHashMatchMachineElectionHash}
               onPress={unprogramCard}
               primary={!doesCardElectionHashMatchMachineElectionHash}
             >
               Unprogram Card
             </Button>
-          </p>
-        </React.Fragment>
+          )}
+        </p>
+      )}
+      {(actionStatus || possibleActions.size > 0) && <HorizontalRule />}
+
+      {showingSuccessOrErrorMessage ? (
+        <Text bold>Remove card to continue.</Text>
+      ) : (
+        <p>
+          Remove card to{' '}
+          {possibleActions.size > 0 ? 'cancel' : 'leave this screen'}.
+        </p>
```

Comment:
This screen is getting a bit complex. It's hard to puzzle out what will actually be rendered. I wonder if it's possible to simplify this logic at all.

Are there a few high level cases that could be broken out into separate `return` blocks (even if there's some repetition)?

Reply:
I agree with Jonah. Don't know if it helps, but when determining how to write programming logic, my goal is to make it easy for another programmer to read. Often this other programmer is me coming back to the code having lost all previous context. :ton

Reply:
Good callout! I agree that the branching within this component has gotten high and that this component could benefit from higher-level branching. Will tweak 🛠️

> when determining how to write programming logic, my goal is to make it easy for another programmer to read. Often this other programmer is me coming back to the code having lost all previous context

Yes, this is a given! I've probably just gotten too familiar with this code over the past few days 😝

Reply:
Okay! Landed on this: https://github.com/votingworks/vxsuite/pull/2267/commits/c983f0ea3b0a01fb90efe24e17194c746c29b357

Hopefully makes what's displayed under different circumstances easier to glean

I know I bias toward avoiding repetition, so if this doesn't go far enough, and you'd still prefer some earlier returns, more than happy to tweak further